torch/csrc/Dtype.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <Python.h>
+#include "ATen/ATen.h"
+
+const int DTYPE_NAME_LEN = 64;
+
+struct THPDtype {
+  PyObject_HEAD
+  at::Type *cdata;
+  char name[DTYPE_NAME_LEN + 1];
+  bool is_cuda;
+  bool is_sparse;
+};
+
+extern PyObject *THPDtypeClass;
```

Comment:
This seems unnecessary to me. Use `extern PyTypeObject* THPDtypeType` instead. Then you don't have to worry about initializing it.

The reason we use the separate class/Type pattern in other places is because we subclass `_VariableBase` with `Variable` in Python, but want to construct the subclass from C++. Since there's no subclass of THPDtypeType here it doesn't matter.